can-event.js

```diff
@@ -66,73 +66,87 @@ var canEvent = {
-    			handlers.splice(i, 1);
-    		} else {
-    			i++;
-    		}
-    	}
-    	return this;
-    },
-    // ## can-event.dispatch
-    //
-    // Dispatches/triggers a basic event on an object.
-    /**
-     * @function can-event.dispatch dispatch
-     * @parent can-event.static
-     * @signature `obj.dispatch(event, [args])`
-     *
-     * Dispatches/triggers a basic event on an object.
+	 *
+	 * @param {String} event The name of the event to listen for.
+	 * @param {Function} handler The handler that will be executed to handle the event.
+	 * @return {Object} this
+	 *
+	 * @signature `canEvent.addEventListener.call(obj, event, handler)`
+	 *
+	 * This syntax can be used for objects that don't include the `canEvent` mixin.
+	 */
+	addEventListener: function (event, handler) {
+		// Initialize event cache.
+		var allEvents = this.__bindEvents || (this.__bindEvents = {}),
+			eventList = allEvents[event] || (allEvents[event] = []);
+
+		// Add the event
+		eventList.push(handler);
+		return this;
+	},
+
+	// ## can-event.removeEventListener
+	//
+	// Removes a basic event listener from an object.
+	// This removes event handlers from the cache of listened events.
+	/**
+	 * @function can-event.removeEventListener removeEventListener
+	 * @parent can-event.static
+	 * @signature `obj.removeEventListener(event, handler)`
+	 *
+	 * Removes a basic event listener from an object. Don't pass arguments to remove all event listeners.
+	 *
+	 * @param {String} event The name of the event to listen for.
+	 * @param {Function} handler The handler that will be executed to handle the event.
+	 * @return {Object} this
+	 *
+	 * @signature `canEvent.removeEventListener.call(obj, event, handler)`
+	 *
+	 * This syntax can be used for objects that don't include the [can-event] mixin.
+	 */
+	removeEventListener: function (event, fn) {
+		if (!this.__bindEvents) {
+			return this;
+		}
+
+		// Remove all eventListeners if no arguments are passed
+		if (!arguments.length) {
+			for (var bindEvent in this.__bindEvents) {
+				if (bindEvent === '_lifecycleBindings') {
+					//reset lifecycleBindings count for lifecycle events
+					this.__bindEvents._lifecycleBindings = null;
+				} else if (this.__bindEvents.hasOwnProperty(bindEvent)) {
+					canEvent.removeEventListener.call(this, bindEvent, fn);
+				}
+			}
+			return this;
+		}
+
+		var handlers = this.__bindEvents[event] || [],
+			i = 0,
+			handler, isFunction = typeof fn === 'function';
+		while (i < handlers.length) {
+			handler = handlers[i];
+			// Determine whether this event handler is "equivalent" to the one requested
+			// Generally this requires the same event/function, but a validation function
+			// can be included for extra conditions. This is used in some plugins like `can/event/namespace`.
+			if ( isFunction && handler === fn || !isFunction && (handler.cid === fn || !fn)) {
+				handlers.splice(i, 1); // likely not using delete for perf reasons
```

Comment:
This is so it willremove a single function if call it like `canEvent.removeEventListener.call(obj, event, handler)`. We could do it the way you're proposing if `handler` isn't passed, but this isn't very performance critical code, so it wouldn't make a big difference I don't think.